### PR TITLE
Update inquirer dependency

### PR DESCRIPTION
**Pull request by btecu**
Also fix tests and typos. For #3670.

### CODE REVIEW

File: tests/unit/models/blueprint-test.js [hunk 1]
```diff
@@ -285,8 +285,8 @@ describe('Blueprint', function() {
         });
     });
 
-    it('re-installing conflicting files', function() {
-      return blueprint.install(options)
+    it('re-installing conflicting files', function(done) {
+      blueprint.install(options)
```
Comment (stefanpenner) [inline tests/unit/models/blueprint-test.js]: why does returning the promise not work?

File: tests/unit/models/blueprint-test.js [hunk 2]
```diff
@@ -377,8 +379,8 @@ describe('Blueprint', function() {
         Blueprint.ignoredUpdateFiles.push('foo.txt');
       });
 
-      it('ignores files in ignoredUpdateFiles', function() {
-        return blueprint.install(options)
+      it('ignores files in ignoredUpdateFiles', function(done) {
+        blueprint.install(options)
```
Comment (stefanpenner) [inline tests/unit/models/blueprint-test.js]: ^^

File: tests/unit/models/blueprint-test.js (hunk 1 above)
Comment (btecu) [inline tests/unit/models/blueprint-test.js]: It causes the test to fail:
```
1) Blueprint basic blueprint installation re-installing conflicting files:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
      at null.<anonymous> (/usr/local/lib/node_modules/mocha/lib/runnable.js:170:19)
      at Timer.listOnTimeout (timers.js:88:15)
```

Comment (stefanpenner) [inline tests/unit/models/blueprint-test.js]: We should figure out why and fix it, without this we will have annoyingly brittle tests

Comment (btecu) [inline tests/unit/models/blueprint-test.js]: Weird, the tests are passing for me locally.

**Comment by stefanpenner** [comment]
I would prefer to continue to use the promise interface for these tests, rather then the `done` approach.

**Comment by btecu** [comment]
How about calling `done` before returning the promise?

**Comment by stefanpenner** [comment]
The promise technique ensures errors and async are fully integrated with mocha, although sometimes more work, it results in more stability of our tests. If your blocked. Let me know I can investigate

File: tests/unit/models/blueprint-test.js [hunk 3]
```diff
@@ -285,9 +285,12 @@ describe('Blueprint', function() {
         });
     });
 
-    it('re-installing conflicting files', function() {
+    it('re-installing conflicting files', function(done) {
       return blueprint.install(options)
         .then(function() {
+          done();
```
Comment (stefanpenner) [inline tests/unit/models/blueprint-test.js]: this is totally in the wrong place

File: tests/unit/models/blueprint-test.js [hunk 4]
```diff
@@ -377,9 +380,12 @@ describe('Blueprint', function() {
         Blueprint.ignoredUpdateFiles.push('foo.txt');
       });
 
-      it('ignores files in ignoredUpdateFiles', function() {
+      it('ignores files in ignoredUpdateFiles', function(done) {
         return blueprint.install(options)
           .then(function() {
+            done();
```
Comment (stefanpenner) [inline tests/unit/models/blueprint-test.js]: ^ same as above

File: tests/unit/models/file-info-test.js [hunk 5]
```diff
@@ -96,46 +96,52 @@ describe('Unit - FileInfo', function(){
...
     });
 
-    return fileInfo.confirmOverwrite().then(function(action){
+    fileInfo.confirmOverwrite().then(function(action){
       var output = ui.output.trim().split(EOL);
       expect(output.shift()).to.match(/Overwrite.*\?/);
       expect(action).to.equal('overwrite');
     });
+
+    return done();
```
Comment (stefanpenner) [inline tests/unit/models/file-info-test.js]: same as above, this `completes` before the entire `confirmOverwrite` part of the test.

File: tests/unit/models/file-info-test.js [hunk 6]
```diff
@@ -96,46 +96,52 @@ describe('Unit - FileInfo', function(){
...
     });
 
-    return fileInfo.confirmOverwrite().then(function(action){
+    fileInfo.confirmOverwrite().then(function(action){
       var output = ui.output.trim().split(EOL);
       expect(output.shift()).to.match(/Overwrite.*\?/);
       expect(action).to.equal('skip');
     });
+
+    return done();
```
Comment (stefanpenner) [inline tests/unit/models/file-info-test.js]: ^

File: tests/unit/models/file-info-test.js [hunk 7]
```diff
@@ -96,46 +96,52 @@ describe('Unit - FileInfo', function(){
...
     });
 
-    return fileInfo.confirmOverwrite().then(function(action){
+    fileInfo.confirmOverwrite().then(function(action){
       var output = ui.output.trim().split(EOL);
       expect(output.shift()).to.match(/Overwrite.*\?/);
       expect(action).to.equal('diff');
     });
+
+    return done();
```
Comment (stefanpenner) [inline tests/unit/models/file-info-test.js]: ^

File: lib/ui/index.js [hunk 8]
```diff
@@ -4,6 +4,7 @@ var PleasantProgress = require('pleasant-progress');
 var Promise          = require('../ext/promise');
 var EOL              = require('os').EOL;
 var writeError       = require('./write-error');
+var Prompt           = require('inquirer').ui.Prompt;
```
Comment (stefanpenner) [inline lib/ui/index.js]: this is explicitly required lazily so we don't incure the penalty at cli boot for parts of the CLI that do not require inquirer

**Comment by stefanpenner** [comment]
this upgrade is tricky, something changed that is breaking our sorta gnarly test helpers of these code-paths...

**Comment by btecu** [comment]
From what I understand, inquirer requires now to call `done` before finishing. So I don't know what the best place to call it is. Maybe chain it after the current promise?

**Comment by stefanpenner** [comment]
> From what I understand,

what leads you to believe this?

> inquirer requires now to call done before finishing

i believe this is merely a work-around, we should solve the underlying issue.

**Comment by btecu** [comment]
That's what I thought based on the errors I was getting before calling `done`, but I don't have much experience with tests.
What is the underlying issue?

**Comment by stefanpenner** [comment]
> What is the underlying issue?

this needs to be discovered, I have not yet had a chance to dig in.

**Comment by stefanpenner** [comment]
this seems to have gone stale...

**Comment by stefanpenner** [comment]
Would love this to be actually updated, but it will need additional work then this PR has.

**Pull request closed without merging**